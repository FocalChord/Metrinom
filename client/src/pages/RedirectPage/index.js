import React, { useState, useEffect } from "react";
import { Redirect, useParams } from "react-router-dom";
import { CookieManager } from "../../utils/";

const RedirectPage = () => {
    const [redirect, setRedirect] = useState(false);
    const params = useParams();

    useEffect(() => {
        const { token } = params;
        CookieManager.setUserToken(token);
        setRedirect(true);
        // eslint-disable-next-line
    }, []);

    if (redirect) {
        return <Redirect to="/" />;
    }

    return (
        <div className="text-center">
            <header className="min-h-screen flex flex-col items-center justify-center bg-black text-white text-6xl">Redirect</header>
        </div>
    );
};

export default RedirectPage;
