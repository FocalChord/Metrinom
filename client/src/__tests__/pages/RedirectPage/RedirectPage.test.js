import React from "react";
import renderer from "react-test-renderer";
import "regenerator-runtime/runtime.js";
import RedirectPage from "../../../pages/RedirectPage";
import { MemoryRouter } from "react-router-dom";

test("<RedirectPage />", () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <RedirectPage />
            </MemoryRouter>,
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
