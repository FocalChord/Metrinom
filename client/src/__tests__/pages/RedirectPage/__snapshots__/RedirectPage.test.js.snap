// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RedirectPage /> 1`] = `
<div
  className="text-center"
>
  <header
    className="min-h-screen flex flex-col items-center justify-center bg-black text-white text-6xl"
  >
    Redirect
  </header>
</div>
`;
